modules/dkan/dkan_datastore/modules/dkan_datastore_api/dkan_datastore_api.module

```diff
@@ -1123,3 +1123,180 @@ function dkan_datastore_api_debug($data_select) {
+      }
+
+      $node = _services_arg_value($request, 'node');
+
+      if (!empty($node['type']) && $node['type'] != $old_node->type) {
+        // Node types cannot be changed once they are created.
+        return services_error(t('Node type cannot be changed'), 406);
+      }
+
+      foreach ($node as $field => $value) {
+        $old_node->{$field} = $value;
+      }
+
+      try {
+        field_attach_validate('node', $old_node);
+        _dkan_datastore_api_field_validation($old_node);
+      }
+      catch (\Exception $e) {
+        return services_error($e->getMessage());
+      }
+
+      node_save($old_node);
+
+      $nid = $old_node->nid;
+      $response = array('nid' => $nid);
+      if ($uri = services_resource_uri(array('node', $nid))) {
+        $response['uri'] = $uri;
+      }
+      return $response;
+    }
+    else {
+      return _node_resource_update($nid, $request);
+    }
+  }
+  else {
+    return services_error("Type not set for node");
+  }
+}
+
+/**
+ * Basic validation of fields.
+ *
+ * Currently we are only checking for required fields, and tha the
+ * values in an options/select list field are valid.
+ */
+function _dkan_datastore_api_field_validation($node) {
+  $instances = field_info_instances('node', $node->type);
+
+  foreach ($instances as $field_name => $info) {
+    if ($info['required'] == 1 && empty($node->{$field_name})) {
+      throw new Exception("Field {$field_name} is required");
+    }
+
+    if (!empty($node->{$field_name})) {
+      $options = _dkan_datastore_api_field_validation_get_options($node, $field_name, $info);
+
+      // Check that the valuess match the options.
+      if (!empty($options) && is_array($options)) {
+        _dkan_datastore_api_field_validation_check_field_values_against_options($field_name, $node->{$field_name}, $options);
+      }
+    }
+  }
+}
+
+/**
+ * Get options for a field.
+ */
+function _dkan_datastore_api_field_validation_get_options($node, $field_name, $field_instance) {
+  $options = [];
+
+  // Look for predefined options for this field.
+  if (!empty($field_instance['widget']['settings']['available_options'])) {
+    $string = $field_instance['widget']['settings']['available_options'];
+    $lines = explode(PHP_EOL, $string);
+    foreach ($lines as $line) {
+      $pieces = explode("|", $line);
+      $options[trim($pieces[0])] = trim($pieces[1]);
+    }
+  }
+  else {
+    $field = field_info_field($field_name);
+    $properties = [
+      'strip_tags' => FALSE,
+      'strip_tags_and_unescape' => FALSE,
+      'filter_xss' => FALSE,
+      'empty_option' => FALSE,
+      'optgroups' => FALSE
+    ];
+
+    $options = _options_get_options($field, $field_instance, $properties, 'node', $node);
+  }
+
+  return $options;
+}
+
+/**
+ * Check field values against field options.
+ */
+function _dkan_datastore_api_field_validation_check_field_values_against_options($field_name, $field_values, $options) {
```

Comment:
Does not check if cardinality is correct - could have submitted multiple values for a single value field. What would happen in that case?

Reply:
Good question. My guess is that it will still work, but I think that checking for cardinality is a good check.